graph/coloring/graph.go

```diff
@@ -0,0 +1,53 @@
+package coloring
+
+import "errors"
+
+// Color provides a type for vertex color
+type Color int
+
+// Graph provides a structure to store an undirected graph.
+// It is safe to use its empty object.
+type Graph struct {
```

Comment:
Is it not possible to put it in the `graph` package instead of adding it inside the `coloring` package?

Reply:
On second thought, I think we should avoid this, doing so will make us convert all the functions from `Graph` methods to functions. And I think we should stick to the current implementation. 

We can copy the `graph.go` file to the `graph` package for others to use. 

GoProverb: "A little copying is better..."

Reply:
What do you think?

Reply:
Yeah, that sounds reasonable. In fact, if you add the implementation then you can use your methods as functions too - just pass it as first argument in your function call. Although I don't think that style should be promoted. Anyways, add the Linear undirected graph to graph package and mark this as resolved😄